File: common/src/main/java/net/nebular/Nebular.java
package net.nebular;

import net.minecraft.resources.ResourceLocation;
import net.nebular.foundation.NebularRegistrate;
import net.nebular.reg.NebularBlockEntities;
import net.nebular.reg.NebularBlocks;
import net.nebular.reg.NebularItems;
import org.slf4j.Logger;
import org.slf4j.LoggerFactory;

public class Nebular {
    public static final String MOD_ID = "nebular";
    public static final String NAME = "Nebular";


    public static final Logger LOGGER = LoggerFactory.getLogger(NAME);
    public static NebularRegistrate REGISTRATE = new NebularRegistrate(MOD_ID);

    public static void init() {
        NebularItems.register();
        NebularBlocks.register();
        NebularBlockEntities.register();

        REGISTRATE.register();
    }

    public static ResourceLocation asResource(String path) {
        return new ResourceLocation(MOD_ID, path);
    }

}

File: common/src/main/kotlin/net/nebular/Nebular.kt
package net.nebular

import com.simibubi.create.foundation.data.CreateRegistrate
import net.minecraft.resources.ResourceLocation
import net.nebular.foundation.NebularRegistrate
import net.nebular.reg.NebularBlocks
import net.nebular.reg.NebularCreativeTabs
import net.nebular.reg.NebularCreativeTabs.BASE
import org.slf4j.Logger
import org.slf4j.LoggerFactory

object Nebular {
    const val MOD_ID = "nebular"
    const val NAME = "Nebular"

    @JvmStatic
    val LOGGER: Logger = LoggerFactory.getLogger(NAME)
    val REGISTRATE: NebularRegistrate = NebularRegistrate(MOD_ID)

    @JvmStatic
    fun init() {
        NebularBlocks.register()
        NebularCreativeTabs.register()

        REGISTRATE.register()
    }

    fun asResource(path: String): ResourceLocation =
        ResourceLocation(MOD_ID, path)

}